src/java.base/share/classes/java/lang/Class.java

```diff
@@ -3491,28 +3484,22 @@ private ReflectionData<T> newReflectionData(SoftReference<ReflectionData<T>> old
     private native String getGenericSignature0();
 
     // Generic info repository; lazily initialized
-    private transient volatile ClassRepository genericInfo;
-
-    // accessor for factory
-    private GenericsFactory getFactory() {
-        // create scope and factory
-        return CoreReflectionFactory.make(this, ClassScope.make(this));
-    }
+    private transient volatile @Stable ClassGenericInfo<T> genericInfo;
```

Comment:
I don’t think this field can be `@Stable`, as generic signatures can change when a class gets redefined by **JVM TI**, just like [simple][Class​::getSimpleName()] and [canonical names][Class​::getCanonicalName()].
https://github.com/openjdk/jdk/blob/8acdd2d7c8de17515b87815d54ce556237039406/src/java.base/share/classes/java/lang/Class.java#L3452-L3454

[Class​::getSimpleName()]: https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/lang/Class.html#getSimpleName()
[Class​::getCanonicalName()]: https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/lang/Class.html#getCanonicalName()

Reply:
Existing genericInfo is not part of ReflectionData. The migration of genericInfo into reflectionData should be done in a separate task.